base/linalg/diagonal.jl

```diff
@@ -101,9 +101,9 @@ end
 parent(D::Diagonal) = D.diag
 
 ishermitian{T<:Real}(D::Diagonal{T}) = true
-ishermitian(D::Diagonal) = all(D.diag .== real(D.diag))
+ishermitian(D::Diagonal) = isreal(D.diag)
```

Comment:
By that argument, maybe I should ditch `iszero(itr)`, since there is no meaningful `zero(itr)`.